From 0865894f23ffea3ee96b7fa73e61db5f5a4e72ed Mon Sep 17 00:00:00 2001
From: Pritesh Patel <pritesh.patel@einfochips.com>
Date: Tue, 30 Jul 2024 14:17:25 +0000
Subject: [PATCH 026/128] drivers: kvm: vcpu: Disabled writing HENVCFG reg

HENVCFG register is not present in EIC7700 SOC. So disabled writing
this register

Signed-off-by: Pritesh Patel <pritesh.patel@einfochips.com>
---
 arch/riscv/kvm/vcpu.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/riscv/kvm/vcpu.c b/arch/riscv/kvm/vcpu.c
index 82229db1ce73..9f1c1f887da2 100644
--- a/arch/riscv/kvm/vcpu.c
+++ b/arch/riscv/kvm/vcpu.c
@@ -487,7 +487,10 @@ static void kvm_riscv_vcpu_update_config(const unsigned long *isa)
 	if (riscv_isa_extension_available(isa, ZICBOZ))
 		henvcfg |= ENVCFG_CBZE;
 
+#ifndef CONFIG_SOC_SIFIVE_EIC7700
 	csr_write(CSR_HENVCFG, henvcfg);
+#endif
+
 #ifdef CONFIG_32BIT
 	csr_write(CSR_HENVCFGH, henvcfg >> 32);
 #endif
-- 
2.47.0

